Library/Homebrew/requirements/macos_requirement.rb

```diff
@@ -0,0 +1,26 @@
+require "requirement"
+
+class MacOSRequirement < Requirement
+  fatal true
+
+  def initialize(tags = [])
+    @version = MacOS::Version.from_symbol tags.first unless tags.empty?
+    super
+  end
+
+  satisfy(build_env: false) do
+    next false if !OS.mac? && !@version
+    next true if !OS.mac? || !@version
```

Comment:
This is a little hard to parse above. Can you maybe make it `if`/`else`s? This may contradict what I said earlier: sorry!

Reply:
Factoring out the `minimum_version?` utility method also helped make this logic more clear.